### DISCUSSION TITLE
How can I get Meson to recognize where a media file is located in a given project

### DISCUSSION THREAD

**Discussion by dreamer-coding** [mesonbuild/meson#13794] [Q&A]
How can I get Meson to recognize where a media file is located in a given project? The goal would be to allow say for example a game or a utility to find a media file it needs to perform some action without having to input the full path to the media file manually nor make use of file config method for copying the files due to being deprecated in previous versions of Meson and prefers users to make use of the file system module.

**Current build file in media directory:**

https://github.com/dreamer-coding/fossil-project/blob/main/code/media/meson.build

Follow up question, if it is possible to have something setup so I can have a set of media files for a given project and a set of media files for unit testing purposes that would be great. 

As a note the media file itself could be a text file, could be an image file or something else, for simplicity I will refer to it as a media file which is basically a file Meson build or any other build tools shouldn't normally be concerned about due to how enlivenment it is when compiling a program with source files being the main focus.

**Comment by eli-schwartz** [comment]
There are various tactics for this:
- checking for it relative to the current binary, using nonstandard platform-specific approaches to find "the current executable's absolute path" (see e.g. `QCoreApplication::applicationFilePath`)
- having an application-specific environment variable for overriding the path, e.g. something you can define via `meson.add_devenv()`)
- accept a command-line argument such as `--media-file=$PWD/my-data-file`

**Reply by dreamer-coding** [reply]
On the application side it give me issues before when I added a group of media files. As for the test side of things I suppose I can give the `devenv` a try.
